src/devices/src/virtio/vsock/device.rs

```diff
@@ -148,7 +148,14 @@ where
                 Ok(mut pkt) => {
                     if self.backend.recv_pkt(&mut pkt, mem).is_ok() {
                         match pkt.commit_hdr(mem) {
-                            Ok(()) => VSOCK_PKT_HDR_SIZE as u32 + pkt.len(),
+                            Ok(()) => pkt.len().checked_add(VSOCK_PKT_HDR_SIZE as u32).unwrap_or_else(|| {
+                                warn!(
+                                    "vsock: packet length greater than maximum expected size: {} > {}",
```

Comment:
I would also say that we're discarding the package.

Reply:
And I would turn the `warn!` into `error!` since we're actually dropping the package.

Reply:
I am not sure if `error!` would be the right level here. All other package drops have been signaled as `warn!` so far.